src/App.tsx

```diff
@@ -1,74 +1,132 @@
-import React from 'react';
+import React, { useState, useMemo, useCallback, useRef } from 'react';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [delay, setDelay] = useState(300);
+  const applyQuery = useCallback(debounce(setQuery, delay), []);
+  const inputRef = useRef(null);
+  const [isSelected, setSelected] = useState(false);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [personInfo, setPersonInfo] = useState<Person | null>(null);
+  const normalizeQuery = query.toLowerCase();
+  const filteredList = useMemo(() => {
+    return peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(normalizeQuery),
+    );
+  }, [normalizeQuery]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
```

Comment:
Calling both `setQuery` and `applyQuery` here might be redundant. Since `applyQuery` is a debounced version of `setQuery`, you might want to remove `setQuery` from here to rely solely on the debounced function.